### DISCUSSION TITLE
Unable to parse simple markdown Table

### DISCUSSION THREAD

**Discussion by diogolessa** [remarkjs/remark#635] [Q&A] [answered]
Hi all,

I am using `remark` and `remark-html` to parse some content to HTML.

However, it is not parsing tables - they're rendered raw.

It looks like it should be something straight-forward, but it doesn't work for me. Is there any additional settings I need to do in order to make it works?

```
const remark = require("remark");
const html = require("remark-html");

const fileContent = `
    | Property     | Default Value |
    | ------------ | ------------- |
    | value        | \`undefined\`   |
    | get          | \`undefined\`   |
    | set          | \`undefined\`   |
    | writable     | \`false\`       |
    | enumerable   | \`false\`       |
    | configurable | \`false\`       |
`
remark()
   .use(html)
   .process(fileContent, (error, data) => {
        console.log('parse', data.contents)
   });      
```

Output:
![image](https://user-images.githubusercontent.com/5168901/108625325-16353900-7429-11eb-9db9-cdcc07bcee64.png)

I appreciate any help.

Thanks.

**Comment by wooorm** [comment, marked as answer]
https://github.com/remarkjs/remark-gfm

**Reply by diogolessa** [reply]
Thanks!

I just realized it works with `remark` <= 12, but this looks much better.

**Comment by yarynakorduba** [comment]
Hi @wooorm ! First of all, thank you so much for working on this project! This plugin saved me a lot of time.

I just have a question regarding the table / code content. Why is it deleted completely during stripping? Is this somehow unsafe to just render the content as simple text without formatting? Or it was just made for convenience?

Thank you in advance for the answer :)

**Reply by wooorm** [reply]
Heya! Can you expand on what’s stripped? Maybe if you expand a bit on how you’re using remark I can help better?

**Reply by yarynakorduba** [reply]
@wooorm  I use `remark` with `remark-gfm` and `strip-markdown` to remove the formatting from the markdown text (I need to have both formatted text and plain text).

My code for removing of markdown formatting looks like this
```
import remark from 'remark';
import strip from 'strip-markdown';
import remarkGfm from 'remark-gfm';

const stripMarkdown = async (mdText: string): Promise<string> => new Promise((resolve, reject) => {
  const funcName = 'stripMarkdown';

  remark()
    .use(remarkGfm)
    .use(strip, { keep: [] })
    .process(mdText, (error, text) => {
        resolve(String(text));
    });
})
```
The problem I am facing now is that `code` and `table` elements are removed completely together with their content. And I can't find a way to keep the text inside these elements. I know there is a note about this in `strip-markdown` docs. But I just wondered: are code and table elements completely removed on purpose? Is there any way to overcome this?

**Reply by wooorm** [reply]
> are code and table elements completely removed on purpose

Yes, block level (flow) code is completely removed on purpose. Same for tables.

> I need to have both formatted text [...]

Do you mean HTML?

**Reply by yarynakorduba** [reply]
@wooorm 
> Do you mean HTML?

no no, just plain text like for the other elements..

> Yes, block level (flow) code is completely removed on purpose. Same for tables.

is this considered unsafe to keep this text?

**Reply by wooorm** [reply]
> is this considered unsafe to keep this text?

This package is more about removing syntax, so that only plain text prose is left. Not about security. And there isn’t a representation for tables or lots of lines of code in plain text.

What type of “plain text” do you expect for your markdown then? Can you show input/output examples?

**Reply by yarynakorduba** [reply]
@wooorm ok, I see
By "plain text" I was meaning something like converting this
\```
code
\```
to this: "code" instead of empty string

or this:
| Tables        | Are           | Cool  |
| ------------- |:-------------:| -----:|
| col 3 is      | right-aligned | $1600 |

to this: "Tables\nAre\nCool\ncol 3 is\nright-aligned\n$1600" instead of empty string

**Reply by yarynakorduba** [reply]
@wooorm but anyway, now that I see there is no ready-to-use solution for this particular use case, I guess the easiest option for me would be to create a fork and adjust it to my need.
Thank you very much for your time and help! :)

**Reply by wooorm** [reply]
@yarynakorduba You can use more backticks (such as 4) around code that includes 3 backticks!

You can indeed fork. Alternatively, HTML has much smarted text algorithms: you could go remark -> rehype and then use `hast-util-to-text` to get “plain text” according to the DOM’s `.textContent` prop.

**Reply by yarynakorduba** [reply]
@wooorm ok, thanks!
